docs/migration_guide.md

```diff
@@ -0,0 +1,51 @@
+# Migrating from scikit-build
+
+```{warning}
+scikit-build-core is under active development.
+This guidance will be updated on a best-effort basis, but if you are working at the bleeding edge some of it may be out of date.
+```
+
+## Config changes
+
+- The `build-system.build-backend` key in pyproject.toml must be changed to
+  `scikit_build_core.build`.
+- Replace `scikit-build` with `scikit-build-core` in `build-system.requires`.
+- You may remove `cmake` and `ninja` from `build-system.requires` (IS THIS
+  REQUIRED? CAN THESE BE LEFT IN PLACE IF DESIRED? WHAT IF STRICTER VERSIONS
+  REQUIREMENTS ARE NEEDED THAN WHAT SCIKIT-BUILD-CORE'S CONFIG SUPPORTS?)
```

Comment:
It is highly, highly recommended since adding them to your requires in the first place is problematic. If you are compiling on a non-binary supported system (which is the main reason a package with lots of wheels gets compiled on a user's system!), then this is broken. That includes WebAssembly, BSDs, Raspberry pi, Android, iOS, SunOS, etc. You can't download a PyPI wheel of ninja or cmake, but you probably can install it yourself. By not putting it here, and letting scikit-build-core inject it if it's required, you can support building from source in a lot more situations.

To configure it, use `[tool.scikit-build] cmake.minimum-required` and `ninja.minimum-required` and `ninja.make-fallback`.